azure/application-gateway/advanced/keyvault.tf

```diff
@@ -11,6 +11,12 @@ resource "azurerm_key_vault" "main" {
   purge_protection_enabled   = true
   sku_name                   = "standard"
   tenant_id                  = data.azurerm_client_config.current.tenant_id
+  lifecycle {
+    precondition {
+      condition     = data.azurerm_client_config.current.tenant_id == data.azurerm_subscription.current.tenant_id
+      error_message = "Your authenticated tenant does not match the tenant where you want to deploy. Please ensure that you are at least a guest user in the tenant where you want to deploy AppGW"
+    }
+  }
 }
 
 resource "azurerm_key_vault_access_policy" "main_user" {
```

Comment:
Does the identity need all those permissions.  e.g - Backup, Purge,

Reply:
Probably not, outside of the scope of the original PR but good spot regardless. Any suggestions on the permissions I could use here?

Reply:
Correct me if I'm wrong but it looks like only a certificate is being created.  So I'd suggest the following:
certificate_permissions = ["Create", "Get", "List", "Update", "Recover, "Restore"]

Reply:
I've reviewed this further this morning and it seems like it also needs "Delete" to run terraform destroy